### PR TITLE
minimum viable product

### PR DESCRIPTION
**Pull request by ben-spector-cc**
Some design patterns that I'm using that I'd love to hear your thoughts on:

* I'm using a constant naming pattern for the message types. The pattern is `SOURCE_MODE.TYPE`.
  * `SOURCE` can be either `CLIENT` or `SERVER`
  * `MODE` can be either `MESSAGE` (1:1) or `BROADCAST`(1:many) 
  * `TYPE` will be unique to the message type 
  * For example: `SERVER_BROADCAST.COUNTDOWN` or `CLIENT_MESSAGE.NEW_USER`

* To implement custom messages, I send all data between the server and client as a `JSON.stringify`-ed object containing a required `.type` property. If the message needs to include a payload, there's an optional `.payload` property that I have consistently made an object containing the data, even if the data is a single primitive value. 
  * For example: `JSON.stringify({ type: CLIENT_MESSAGE.NEW_USER, payload: { name: 'Ben' } })`

Aside from these patterns, any feedback you have on the WebSockets implementation, the application structure, and the application's organization would be great. 

And for now, please focus your feedback on the parts of the application related to what the user will have to implement. Please ignore the DOM logic and we can circle back to it once we get the final designs in from our designer.

Thanks!

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+  <h1 id='display'>Waiting for game to start...</h1>
+  <div id="table">
+    <div class='player' id='player1'></div>
+    <div class='player' id='player2'></div>
+    <div class='player' id='player3'></div>
+    <div class='player' id='player4'></div>
+  </div>
+    
+  <script>
+    (function() {
```
Comment (Ethan-Arrowood) [inline public/index.html]: iife is not necessary
```suggestion

```

File: public/index.html [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+        // .onerror is executed when error event occurs on the WebSocket connection
+        ws.onerror = function(e) {
+          console.error("WebSocket error observed:", e);
+          ws = null;
+        }
+      }
+  
+      init();
+    })();
```
Comment (Ethan-Arrowood) [inline public/index.html]:
```suggestion

```

File: server.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  PLAYER_ASSIGNMENT: 'PLAYER_ASSIGNMENT',
+  GAME_FULL: 'GAME_FULL',
+};
+const SERVER_BROADCAST = {
+  COUNTDOWN: 'COUNTDOWN',
+  NEW_POTATO_HOLDER: 'NEW_POTATO_HOLDER',
+  GAME_OVER: 'GAME_OVER',
+  GAME_START: 'GAME_START',
+}
+
```
Comment (Ethan-Arrowood) [inline server.js:9]:
```suggestion
const CONSTANTS = {
  CLIENT: {
    MESSAGE: {
      NEW_USER: 'new user',
      PASS_POTATO: 'pass potato'
    }
  },
  SERVER: {
    MESSAGE: {
      PLAYER_ASSIGNMENT: 'player assignment',
      GAME_FULL: 'game full',
    },
    BROADCAST: {
      COUNTDOWN: 'COUNTDOWN',
      NEW_POTATO_HOLDER: 'NEW_POTATO_HOLDER',
      GAME_OVER: 'GAME_OVER',
      GAME_START: 'GAME_START',
    }
  }
}
```

File: server.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+// Application Variables;
+let players = [];
```
Comment (Ethan-Arrowood) [inline server.js]:
```suggestion
const players = [];
```

**Review by Ethan-Arrowood — CHANGES_REQUESTED** [review]

**Comment by ben-spector-cc** [comment]
@Ethan-Arrowood Just moved the constants over into their own file and removed the IIFE. I also refactored the HTTP server request/response handler to be a little bit cleaner.

**Review by Ethan-Arrowood — APPROVED** [review]
lgtm

**Pull request merged**